File: Tarefeiro.Solution/Tarefeiro.Application/DTOs/Tarefa/TarefaDto.cs
using System;

namespace Tarefeiro.Application.DTOs.Tarefa;
public record class ShortStatusDto(int id, string Status);
public record class ShortCategoriaDto(int id, string Categoria);
public class TarefaDto
{
    public int Id { get; set; }
    public ShortStatusDto Status { get; set; } = null!;
    public ShortCategoriaDto Categoria { get; set; } = null!;
    public string Comentario { get; set; } = string.Empty;
}

File: Tarefeiro.Front/src/app/features/components/tarefas/models/tarefa-dto.ts
export interface ShortStatusDto {
    id: number;
    status: string;
}

export interface ShortCategoriaDto {
    id: number;
    categoria: string;
}

export interface TarefaDto {
    id: number;
    comentario: string;
    status: ShortStatusDto;
    categoria: ShortCategoriaDto;
}